### DISCUSSION TITLE
等待一个二进制版🤩🤩🤩

### DISCUSSION THREAD

**Discussion by lhuanyun** [usememos/memos#340] [Q&A]
非常好的记录器，有二进制版本方便部署迁移。docker 上面的项目不想太零碎了，谢谢。

**Comment by Zeng1998** [comment]
> docker 上面的项目不想太零碎了

（不太理解这是什么强迫症

二进制版也同样还得引入其他工具去管理

**Comment by lhuanyun** [comment]
OS :debian 11.4
---
创建 /etc/rc.local
其中写入:
```
nohup /mnt/app/portainer-2.5.0-linux-amd64/portainer --data /mnt/app/portainer-2.5.0-linux-amd64/portainer-data  -p:63878 --tunnel-port="8100" >>/mnt/app/DebianAppLog 2>&1 &
```
这样的话 portainer-2.5.0就会在开机的时候自动后台运行。
好处是，复制一下/mnt/app/portainer-2.5.0-linux-amd64这个文件夹，放到哪里。都可以用命令快速启动，全部资料都迁移走。

**Comment by Zeng1998** [comment]
> ## OS :debian 11.4
> 创建 /etc/rc.local 其中写入:
> 
> ```
> nohup /mnt/app/portainer-2.5.0-linux-amd64/portainer --data /mnt/app/portainer-2.5.0-linux-amd64/portainer-data  -p:63878 --tunnel-port="8100" >>/mnt/app/DebianAppLog 2>&1 &
> ```
> 
> 这样的话 portainer-2.5.0就会在开机的时候自动后台运行。 好处是，复制一下/mnt/app/portainer-2.5.0-linux-amd64这个文件夹，放到哪里。都可以用命令快速启动，全部资料都迁移走。

好吧，挺炫酷的，不过仍然感觉不是一个特别实际的需求...

**Comment by lhuanyun** [comment]
> 
> 
> 
> 
> 
> 
> > ## OS :debian 11.4
> 
> > 创建 /etc/rc.local 其中写入:
> 
> > 
> 
> > ```
> 
> > nohup /mnt/app/portainer-2.5.0-linux-amd64/portainer --data /mnt/app/portainer-2.5.0-linux-amd64/portainer-data  -p:63878 --tunnel-port="8100" >>/mnt/app/DebianAppLog 2>&1 &
> 
> > ```
> 
> > 
> 
> > 这样的话 portainer-2.5.0就会在开机的时候自动后台运行。 好处是，复制一下/mnt/app/portainer-2.5.0-linux-amd64这个文件夹，放到哪里。都可以用命令快速启动，全部资料都迁移走。
> 
> 
> 
> 好吧，挺炫酷的，不过仍然感觉不是一个特别实际的需求...

🤔我个人使用 倒是觉得很实用
rc.local 等于启动菜单，直观。
不需要启动，就把命令注释掉。
![image](https://user-images.githubusercontent.com/20707383/197658922-e9c915a3-3afd-4c39-a290-04683bdc1f66.png)

配合手机进行启动关闭程序服务。
![image](https://user-images.githubusercontent.com/20707383/197659005-52a214f3-f119-4e78-8228-90621884605a.png)

**Comment by Zeng1998** [comment]
那这样子跟docker在哪个地方会有很大区别吗？

**Comment by lhuanyun** [comment]
> 那这样子跟docker在哪个地方会有很大区别吗？

go可以独立运行，不需要docker。
数据迁移保存比docker持久方便很多，整体转移不会出问题。😕

**Comment by boojack** [comment]
Related discussion: https://github.com/usememos/memos/discussions/335

You can clone this repo and build it on your machine.

**Comment by lhuanyun** [comment]
😣😣😣
不会nodejs

**Comment by Zeng1998** [comment]
用docker，数据转移也只是copy一个db文件就行的

**Comment by anno189** [comment]
[memos 0.6.1.zip](https://github.com/usememos/memos/files/9858416/memos.0.6.1.zip)

我自己编译的，希望你能用。

**Comment by lhuanyun** [comment]
> [memos 0.6.1.zip](https://github.com/usememos/memos/files/9858416/memos.0.6.1.zip)
> 
> 
> 
> 我自己编译的，希望你能用。

![image](https://user-images.githubusercontent.com/20707383/197723304-5f687e8e-a4ac-4174-8dc5-280686ab4dd0.jpeg)
可以工作。 可以请教一下，编译流程吗

**Reply by fengqi** [reply]
参考这个 https://github.com/usememos/memos/blob/main/Dockerfile 安装node、yarn、go，运行两个 build

**Comment by LittleArray** [comment]
我用Action编译完了基于0.12.2的memos
下载地址：[Link](https://github.com/LittleArray/memos-action/actions/runs/4761944331)
往下翻找到对应平台的下载
## 环境配置
请在运行目录下新建`\var\opt\memos`的目录结构

## 运行命令
### Windows
`memos.exe --port 80 --mode pord`
### Linux
`memos --port 80 --mode pord`

**Reply by guanzi008** [reply]
> 我用Action编译完了基于0.12.2的memos 下载地址：[Link](https://github.com/LittleArray/memos-action/actions/runs/4761944331) 往下翻找到对应平台的下载
> 
> ## 环境配置
> 请在运行目录下新建的目录结构`\var\opt\memos`
> 
> ## 运行命令
> ### 窗户
> `memos.exe --port 80 --mode pord`
> 
> ### Linux目录
> `memos --port 80 --mode pord`

用的我的代码啊。

**Reply by LittleArray** [reply]
我的这个仓库已经删了，不好意思，我本来是想用来改的，fork了一下
